src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -1371,26 +1368,27 @@ static void window_editor_object_selection_manage_tracks()
  */
 static void editor_load_selected_objects()
 {
+    auto& objManager = OpenRCT2::GetContext()->GetObjectManager();
     int32_t numItems = static_cast<int32_t>(object_repository_get_items_count());
     const ObjectRepositoryItem* items = object_repository_get_items();
     for (int32_t i = 0; i < numItems; i++)
     {
         if (_objectSelectionFlags[i] & OBJECT_SELECTION_FLAG_SELECTED)
         {
-            const ObjectRepositoryItem* item = &items[i];
-            const rct_object_entry* entry = &item->ObjectEntry;
-            const auto* loadedObject = object_manager_get_loaded_object(ObjectEntryDescriptor(*item));
+            const auto* item = &items[i];
+            auto descriptor = ObjectEntryDescriptor(*item);
+            const auto* loadedObject = objManager.GetLoadedObject(descriptor);
             if (loadedObject == nullptr)
             {
-                loadedObject = object_manager_load_object(entry);
+                loadedObject = objManager.LoadObject(descriptor);
                 if (loadedObject == nullptr)
                 {
-                    log_error("Failed to load entry %.8s", entry->name);
+                    log_error("Failed to load entry %s", std::string(descriptor.GetName()).c_str());
```

Comment:
Unsure if this is safe. Lifetime of std::string

Reply:
Lifetime here ends after the call

Reply:
The `std::string` is a temporary and is valid until the end of the full expression, which is after `log_error` gets called, so this is perfectly safe.